src/v/net/probes.cc

```diff
@@ -296,4 +313,251 @@ std::ostream& operator<<(std::ostream& o, const client_probe& p) {
+            auto srl = get_serial(cert);
+            if (
+              _ca_expiry_time == clock_type::time_point{}
+              || exp < _ca_expiry_time) {
+                _ca_expiry_time = exp;
+                _ca_serial = srl;
+            }
+            vlog(log.debug, "_ca_expiry: {}, _ca_serial: {}", exp, to_hex(srl));
+            gnutls_x509_crt_deinit(cert);
+        }
+    }
+
+    // TODO(oren): I suspect this is not needed as the certs should have failed
+    // to load?
+    if (
+      _cert_expiry_time > clock_type::time_point{}
+      && _cert_expiry_time < clock_type::now()) {
+        _cert_valid = false;
+        vlog(
+          log.warn,
+          "Certificate expired at {} (serial: {})",
+          _cert_expiry_time,
+          _cert_serial);
+    } else if (
+      _ca_expiry_time > clock_type::time_point{}
+      && _ca_expiry_time < clock_type::now()) {
+        _cert_valid = false;
+        vlog(
+          log.warn,
+          "Truststore expired at {} (serial: {})",
+          _ca_expiry_time,
+          _ca_serial);
+    }
+}
+
+void tls_certificate_probe::setup_metrics(
+  std::string_view service, std::string_view listener_name) {
+    if (ss::this_shard_id() != 0 || config::shard_local_cfg().disable_metrics) {
+        return;
+    }
+
+    namespace sm = ss::metrics;
+    const auto service_label = sm::label("service");
+    const auto listener_label = sm::label("listener_name");
+    const auto serial_label = sm::label("serial");
+    const auto loaded = sm::label("loaded");
+
+    const std::vector<sm::label_instance> labels = {
+      service_label(service), listener_label(listener_name)};
+
+    const auto aggregate_labels = config::shard_local_cfg().aggregate_metrics()
+                                    ? std::vector<sm::label>{sm::shard_label}
+                                    : std::vector<sm::label>{};
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("tls"),
+      {
+        sm::make_gauge(
+          "truststore_expires_at",
+          [this] {
+              return std::chrono::duration_cast<std::chrono::seconds>(
+                       _ca_expiry_time.time_since_epoch())
+                .count();
+          },
+          sm::description(
+            "Expiry time of the shortest-lived CA in the truststore"
+            "(seconds since epoch)"),
+          labels)
+          .aggregate(aggregate_labels),
+        sm::make_gauge(
+          "certificate_expires_at",
+          [this] {
+              return std::chrono::duration_cast<std::chrono::seconds>(
+                       _cert_expiry_time.time_since_epoch())
+                .count();
+          },
+          sm::description(
+            "Expiry time of the server certificate (seconds since epoch)"),
+          labels)
+          .aggregate(aggregate_labels),
+        sm::make_gauge(
+          "certificate_serial",
+          [this] { return boost::lexical_cast<long>(to_hex(_cert_serial)); },
```

Comment:
>push the work inside probe::loaded

Yeah good call. On a related note, the gnutls man pages indicate:

>This function will return the X.509 certificate's serial number. This is obtained by the X509 Certificate serialNumber field. Serial is not always a 32 or 64bit number. Some CAs use large serial numbers, thus it may be wise to handle it as something uint8_t.

So this all seems a little sketchy anyway...might be better to omit the SN altogether.

Reply:
I've ripped the serial number metric out of the PR and made a note on the PRD. Can add it back if we come to some better conclusion.

Reply:
Put it back per discussion on the PRD - we'll return the least significant 4B of the serial number.